compiler/passes/normalize.cpp

```diff
@@ -1751,14 +1751,15 @@ static void change_method_into_constructor(FnSymbol* fn) {
 
   // Now check that the function name matches the name of the type
   // attached to 'this'.
```

Comment:
I personally think it would be worth it for clarity for us to create and use a stringEquals method. Whether or not you explicitly use == 0 with strcmp, it's obfuscating the logic.

Of course, theoretically, we're not supposed to need to strcmp and you should be able to use pointer equality with astr("init"). I'm not sure I trust that approach though.